pkg/docker/docker.go

```diff
@@ -439,7 +538,7 @@ func (d *stiDocker) CheckAndPullImage(name string) (*docker.Image, error) {
 	}
 
 	image, err := d.CheckImage(name)
-	if err != nil && err.(errors.Error).Details != docker.ErrNoSuchImage {
+	if err != nil && dockerapi.IsErrImageNotFound(err) {
```

Comment:
should you be using the kube api's IsErrImageNotFound, not the docker api one?  is that why you're seeing a difference in the error content?